showcase/tests/integration/modifiers/hds-float-popover-test.js

```diff
@@ -0,0 +1,235 @@
+      });
+      assert.deepEqual(floatingUIOptions.middleware.length, 3);
+      assert.deepEqual(floatingUIOptions.middleware[0].name, 'offset');
+      assert.deepEqual(floatingUIOptions.middleware[1].name, 'flip');
+      assert.deepEqual(floatingUIOptions.middleware[1].options, { padding: 8 });
+      assert.deepEqual(floatingUIOptions.middleware[2].name, 'shift');
+      assert.deepEqual(floatingUIOptions.middleware[2].options, { padding: 8 });
+      floatingUIOptions = getFloatingUIOptions({
+        popoverEnableCollisionDetection: true,
+        popoverFlipOptions: { padding: 1234 },
+        popoverShiftOptions: { padding: 9876 },
+      });
+      assert.deepEqual(floatingUIOptions.middleware[1].options, {
+        padding: 1234,
+      });
+      assert.deepEqual(floatingUIOptions.middleware[2].options, {
+        padding: 9876,
+      });
+    });
+    test('returns the `popoverMiddlewareExtra` provided as an option', async function (assert) {
+      function testFunction() {
+        return { name: 'test', options: { abc: 1234 } };
+      }
+      const floatingUIOptions = getFloatingUIOptions({
+        popoverEnableCollisionDetection: true,
+        popoverMiddlewareExtra: [testFunction()],
+      });
+      assert.deepEqual(floatingUIOptions.middleware.length, 4);
+      // we want also to test the order of the middleware (the extra should be the last)
+      assert.deepEqual(floatingUIOptions.middleware[0].name, 'offset');
+      assert.deepEqual(floatingUIOptions.middleware[1].name, 'flip');
+      assert.deepEqual(floatingUIOptions.middleware[2].name, 'shift');
+      assert.deepEqual(floatingUIOptions.middleware[3].name, 'test');
+      assert.deepEqual(floatingUIOptions.middleware[3].options, { abc: 1234 });
+    });
+    test('returns the `popoverArrow` provided as an option', async function (assert) {
+      const floatingUIOptions = getFloatingUIOptions({
+        // notice: this in reality should be a DOM element, but we found a string works as well for this type of test
+        popoverArrow: 'test1234',
+      });
+      assert.deepEqual(floatingUIOptions.middleware[1].name, 'arrow');
+      assert.deepEqual(floatingUIOptions.middleware[1].options, {
+        element: 'test1234',
+        padding: 8,
+      });
+    });
+  }
+);
+
+// ================================================================
+
+function wait(timeout = 500) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, timeout);
+  });
+}
+
+const hbsPopoverBasicElements = hbs`
+  <div id="wrapper">
+    <div id="toggle">toggle</div>
+    <div id="popover"><div id="arrow"></div>popover</div>
+  </div>
+`;
+
+module('Integration | Modifier | hds-float-popover', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function () {
+    this.styleElement = document.createElement('style');
+
+    // see: https://codepen.io/didoo/pen/VwNpOJb
+    this.styleElement.textContent = `
+      body {
+        padding: 0px;
```

Comment:
Just to mention. Browsers add a margin to the HTML body element by default. They don't add padding.

Reply:
You're right 🤦‍♂️